data_structures/bst.js

```diff
@@ -12,25 +12,107 @@
  */
 
 class BinarySearchTree {
-    constructor(){
-        
+    constructor(value){
+        this.value = value
+        this.right = null
+        this.left = null
     }
     
     //insert the given value into the tree
     insert(value){
-        
+        let new_leaf = new BinarySearchTree(value)
+
+        //NEED TO MAKE SURE THERE IS A RECURSION FOR MORE THAN 2 LEAFS ADDED:
+
+        if(this === null){ 
+            return new_leaf
+        }
+        else{
+            this.insertLeaf(this, new_leaf)
+        }
+        return this
     }
+
+    insertLeaf(tree, newLeaf){
+        if(newLeaf.value < tree.value){
+            if(tree.left === null){
+                tree.left = newLeaf
+            }
+            else{
+                this.insertLeaf(tree.left, newLeaf)
+            }
+        }
+        else{
+            if(tree.right === null){
+                tree.right = newLeaf
+            }
+            else{
+                this.insertLeaf(tree.right, newLeaf)
+            }
+        }
+    }
+
+    
     
     //returns true if the tree contains the value otherwise it returns false
     contains(value){
-        
+        let current_leaf = this
+
+        while(current_leaf !== null){
+            if(current_leaf.value === value){
+                return true
+            }
+            else if(value < current_leaf.value){
+                current_leaf = current_leaf.left
+            }
+            else{
+                current_leaf = current_leaf.right
+            }
+        }
+        return false
     }
     
     //returns the maximum value found in the tree
-    get_max(){}
+    get_max(){
+        if(!this){
+            return null
+        }
+        else if(this.left === null && this.right === null){
+            return this.value
+        }
+        
+        let max_num = Number.MIN_VALUE
```

Comment:
You don't need the max_num here. Since a property of the Binary Search Tree is that the maximum value will be the farthest right node you can just traverse all the way down to the last right node

```javascript
get_max(){
         //base case tree is empty
         if(!this.value){
            return null;
        }
        //go to the right
        let temp = this.right;
        while(temp.right){
            temp = temp.right;
        }
        return temp.value;
    }
```